include/odp/api/spec/packet.h

```diff
@@ -1979,6 +1989,147 @@ int8_t odp_packet_shaper_len_adjust(odp_packet_t pkt);
  */
 void odp_packet_shaper_len_adjust_set(odp_packet_t pkt, int8_t adj);
 
+/*
+ *
+ * Packet vector handling routines
+ * ********************************************************
+ *
+ */
+
+/**
+ * Get packet vector handle from event
+ *
+ * Converts an ODP_EVENT_PACKET_VECTOR type event to a packet vector handle
+ *
+ * @param ev Event handle
+ * @return Packet vector handle
+ *
+ * @see odp_event_type()
+ */
+odp_packet_vect_t odp_packet_vect_from_event(odp_event_t ev);
+
+/**
+ * Convert packet vector handle to event
+ *
+ * @param vpkt Packet vector handle
+ *
+ * @return Event handle
+ */
+odp_event_t odp_packet_vect_to_event(odp_packet_vect_t vpkt);
+
+/**
+ * Allocate a packet vector from a packet vector pool
+ *
+ * Allocates a packet vector from the specified packet vector pool.
+ * The pool must have been created with the ODP_POOL_PACKET_VECTOR type.
+ *
+ * @param pool Packet vector pool handle
+ *
+ * @return Handle of allocated packet vector
+ * @retval ODP_PACKET_VECT_INVALID  Packet vector could not be allocated
+ */
+odp_packet_vect_t odp_packet_vect_alloc(odp_pool_t pool);
+
+/**
+ * Free packet vector
+ *
+ * Frees the packet vector into the packet vector pool it was allocated from.
+ *
+ * @param vpkt Packet vector handle
+ */
+void odp_packet_vect_free(odp_packet_vect_t vpkt);
+
+/**
+ * Get packet vector table
+ *
+ * Packet vector table is a array of packets(odp_packet_t) stored
+ * in contiguous memory location. Upon completion of this API,
+ * the implementation returns the packet table pointer in pkts_tbl.
```

Comment:
Usually, we have get/set functions to read and write event metadata. Since this function differs from that, it should be documented that application can edit packet handles in the table directly (up to max_size) and must set new vector size if number of handles was changed.

Reply:
Ack

Reply:
ownership of the table is linked to the ownership of the event. e.g. after sending the event to a queue, sender looses ownership to the table also.

Reply:
packet handles must be packed into the start of the table (?)